core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -659,6 +659,14 @@ package object config {
       .booleanConf
       .createWithDefault(false)
 
+  private[spark] val SHUFFLE_SERVICE_REMOVE_SHUFFLE_ENABLED =
+    ConfigBuilder("spark.shuffle.service.removeShuffle")
+      .doc("Whether to use the ExternalShuffleService for deleting shuffle blocks for " +
+        "deallocated executors.")
```

Comment:
comment on permissions change.
this description is really unclear to me, and actually so is the title on this.  I think we need to reword and be specific here.  I assume shuffle blocks aren't removed when executor just deallocated, otherwise things would break very quickly. They are only removed when the RDD isn't referenced anymore and context cleaner cleans them up, correct?
This also needs to go into the configuration.md so it shows up in the docs

Reply:
Updated the doc and added it to the configuration.md file as well, let me know how it sounds